src/components/ModalWithConfirmation.js

```diff
@@ -0,0 +1,27 @@
+import Modal from "../components/Modal";
+
+export default class ModalWithConfirmation extends Modal {
+  constructor(modalSelector, handleCardDelete) {
+    super(modalSelector);
+    this._modalForm = this._modalElement.querySelector(".form");
+    this._handleCardDelete = handleCardDelete;
+    this._submitButton = this._modalForm.querySelector(".form__save-button");
+  }
+
+  setEventListeners() {
+    super.setEventListeners();
+    this._modalElement.addEventListener("submit", (e) => {
+      e.preventDefault();
+      this._handleCardDelete(this);
+      this.close();
```

Comment:
Needs correction: Сlose modal in the case of successful api response in `then` block in `index.js`, otherwise the form will close even if an error occurred and you will need to reopen the form